File: presentation/hud.py
from __future__ import annotations
import pygame
import constants


class HUD:
    def __init__(self) -> None:
        self._font = pygame.font.Font(None, 54)

    def draw_pause(self, surface: pygame.Surface) -> None:
        self._draw_centered(surface, "PAUSA", constants.SCREEN_HEIGHT // 2 - 20)
        self._draw_centered_small(surface, "[ P ] continuar", constants.SCREEN_HEIGHT // 2 + 30)

    def _draw_centered(self, surface: pygame.Surface, text: str, y: int) -> None:
        surf = self._font.render(text, True, constants.WHITE)
        surface.blit(surf, (constants.SCREEN_WIDTH // 2 - surf.get_width() // 2, y))

    def _draw_centered_small(self, surface: pygame.Surface, text: str, y: int) -> None:
        font = pygame.font.Font(None, 32)
        surf = font.render(text, True, constants.GRAY)
        surface.blit(surf, (constants.SCREEN_WIDTH // 2 - surf.get_width() // 2, y))

File: presentation/hud.js
import { COLOR_WHITE, COLOR_GRAY, SCREEN_WIDTH, SCREEN_HEIGHT } from './constants.js';

export class HUD {
    constructor(ctx) { this._ctx = ctx; }

    drawPause() {
        const ctx = this._ctx;
        ctx.fillStyle = 'rgba(0,0,0,0.55)';
        ctx.fillRect(0, 0, SCREEN_WIDTH, SCREEN_HEIGHT);
        ctx.textAlign = 'center';
        ctx.fillStyle = COLOR_WHITE;
        ctx.font      = 'bold 56px monospace';
        ctx.fillText('PAUSA', SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2 - 20);
        ctx.fillStyle = COLOR_GRAY;
        ctx.font      = '28px monospace';
        ctx.fillText('[ P ]  continuar', SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2 + 32);
    }
}
